### ISSUE TITLE
NPM_TOKEN not picked up for private dependencies

### ISSUE THREAD

**Issue by fed135** [BePo65/license-report#22]
Getting some errors when repo has private packages

```
http request to npm for package "@****/*****" failed, retrying again soon...
Error: invalid statusCode 404
```

The NPM_TOKEN env variable is set and valid

**Comment by chiko123** [comment]
Solve this as people looking for licenses are really sitting in companies with some private repositories.

**Comment by BePo65** [comment]
Perhaps we should add a configuration variable called 'npmTokenEnvVar' (default value: 'NPM_TOKEN') that contains the name of the environment variable with the bearer token for authentication with the private repo (the address of the repository is defined in the configuration variable 'registry' - default value 'https://registry.npmjs.org/').

Would this solve the issue?

If have made a little test that showed that we can always set the authentication header as the public npm repository simply ignores this header. Unfortunately I don't have access to a private npm repository so that I cannot test this feature.

**Comment by BePo65** [comment]
Fixed in next release based on pr #65.
